src/main/java/org/ambraproject/rhino/view/comment/CommentFlagOutputView.java

```diff
@@ -53,8 +53,10 @@ public JsonElement serialize(JsonSerializationContext context) {
     // Should match the field name used for org.ambraproject.rhino.view.comment.CommentFlagInputView.body.
     serialized.add("body", serialized.remove("comment"));
 
-    serialized.remove("userProfileID");
-    serialized.add("creator", context.serialize(new UserIdView(flag.getUserProfileId())));
+    if (flag.getUserProfileId() != null) {
+      serialized.remove("userProfileID");
```

Comment:
I think this line serialized.remove("userProfileID") should be before if() block.